### ISSUE TITLE
Issue registering input on USB N64 Controller

### ISSUE THREAD

**Issue by danderb** [HarbourMasters/2ship2harkinian#562]
- [x] I have checked the known issues and searched Github to ensure this issue has not already been reported.
- [x] This is not a feature request, I know those belong on Discord/Github Discussions.

### Description of the bug:
Using a retrolink USB Nintendo 64 controller, I am unable to correctly register input.  The Analog stick will always register in the Up Position, unless I manually Hold the stick down 30-45 degrees.  Windows registers the stick at dead center in the controller configuration

### Can you reproduce? If so please list the steps:
- [ ] No, I cannot reproduce this bug
- [x] Yes, here are the steps to reproduce the bug:

1. Go to configure controller
2. Try to set the control scheme
3. See the bug

### Expected behavior:
Control settings should not set until input is provided by user

### System Information:
- OS: Windows 11
- GFX Backend: DirectX
- Game Version: 1.0
- Rom Version: N64 US 1.0

**Comment by danderb** [comment]
Please let me know if additional information can be provided, or if additional debug steps could be helpful.

**Comment by briaguya0** [comment]
could you share your 2ship json? that has device ids in it that can be helpful

**Comment by danderb** [comment]
Attached.
[2ship2harkinian.json](https://github.com/HarbourMasters/2ship2harkinian/files/15470915/2ship2harkinian.json)

**Comment by briaguya0** [comment]
someone on discord had the same issue with the same controller, they were able to fix it given the following information

> ok, did some digging into the sdl changes and it seems they just added a mapping, so the next thing to try would be adding this to your gamecontrollerdb
> ```
> "03000000790000000600000000000000,Sanwa Supply JY-P76USV,crc:20f0,+leftx:+a0,+rightx:+a2,+righty:a4,-leftx:-a0,-lefty:-a1,-rightx:-a2,-righty:-a4,a:b0,b:b1,back:b8,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,leftshoulder:b4,leftstick:b10,lefttrigger:b6,lefty:+a1,rightshoulder:b5,rightstick:b11,righttrigger:b7,start:b9,x:b2,y:b3,",
> ```
> (based on <https://github.com/seyhajin/sdl/commit/e323630043decdc63c9728f948e82841cd73c9e5>)
> 
> it may need to be slightly changed (that doesn't look quite like the gamecontrollerdb formatting)
> 
> according to <https://github.com/libsdl-org/SDL/issues/8644#issuecomment-1982342896>
> > I removed the quotation marks and the last comma, appended ,platform:Windows, put it in gamecontrollerdb.txt and loaded it in my test program. Everything worked correctly.

**Comment by briaguya0** [comment]
this should be fully fixed (no gamecontroller db modification needed) once vcpkg gets SDL 2.30.2

https://github.com/libsdl-org/SDL/releases/tag/release-2.30.2

> * Fixed the gamepad mapping for the Sanwa Supply JY-P76USV controller

**Comment by danderb** [comment]
I have attempted this in my gamecontrollerdb file, but it still appears that the deadzone (specifically in the Y axis) is approximately 20 degrees off center.
[gamecontrollerdb.txt](https://github.com/HarbourMasters/2ship2harkinian/files/15493007/gamecontrollerdb.txt)

**Comment by danderb** [comment]
Also recording of issue attached.  This is with the controller sitting on my desk with no input.

https://github.com/HarbourMasters/2ship2harkinian/assets/153551206/ee8d5a7c-30ce-4b8a-b424-abbf993705f6

**Comment by briaguya0** [comment]
what happens if you calibrate your controller using windows joystick calibration?

**Comment by briaguya0** [comment]
also might be worth trying removing `crc:20f0,` from the entry in the db. i also asked the person who was able to fix it by adding an entry to the `gamecontrollerdb.txt` for the exact string that works for them - so hopefully when i hear back from them we'll have more to work from

edit: it seems the string they added is identical to what you have in the `gamecontrollerdb.txt` file you posted

**Comment by briaguya0** [comment]
one more thing to check/try - after updating the gamecontrollerdb.txt did you try deleting your 2ship json file?

**Comment by danderb** [comment]
deleted crc:20f0 and deleted the 2Ship json file.  Still exhibiting the same behavior.  When i look at windows controller joystick calibration, the stick is dead center at rest, and reacts as expected depending on the direction I move it.

**Comment by briaguya0** [comment]
the person who had the same issue on discord shared the `gamecontrollerdb.txt` file they were using. do you have the same issue when you replace your `gamecontrollerdb.txt` file with this [gamecontrollerdb.txt](https://github.com/user-attachments/files/15528488/gamecontrollerdb.txt)?

**Comment by danderb** [comment]
Replacing my gamecontrollerdb file with the one you provided has made the issue go away.

**Comment by briaguya0** [comment]
thanks for confirming! i'm going to close this issue out for now

> [!NOTE] 
> known issue with this device and SDL2 2.30.1 (the version Rika Bravo pulls from vcpkg)
> it was fixed in 2.30.2 https://github.com/libsdl-org/SDL/releases/tag/release-2.30.2
> > Fixed the gamepad mapping for the Sanwa Supply JY-P76USV controller
>
>  once https://github.com/microsoft/vcpkg/pull/39097 lands vcpkg will pull 2.30.3 and this won't be a problem anymore

**Comment by danderb** [comment]
Appreciate all the help.  All for closing, but until https://github.com/microsoft/vcpkg/pull/39097 occurs, would it be best to package the update gamecontrollerdb.txt file with 2ship?

**Comment by briaguya0** [comment]
We pull that file directly from https://github.com/mdqinc/SDL_GameControllerDB, we've discussed maintaining a fork and pulling from that but that adds a decent amount of overhead. Another option would be someone PRing that mapping string upstream to the db repo. If we're close to release and vcpkg hasn't updated we can look into those options, but considering the vcpkg PR has been approved I expect it to land before the next release of 2ship.

Edit: just checked and the vcpkg PR has been merged
